src/ion.hpp

```diff
@@ -65,14 +65,15 @@ public :
         Xi_{idx.size(), std::numeric_limits<value_type>::quiet_NaN()},
         Xo_{idx.size(), std::numeric_limits<value_type>::quiet_NaN()},
         valency(0),
-        default_internal_concentration(0),
-        default_external_concentration(0)
+        default_int_concentration(0),
+        default_ext_concentration(0)
     {}
 
-    std::size_t memory() const {
-        return 4u*size() * sizeof(value_type)
-               +  size() * sizeof(iarray)
-               +  sizeof(ion);
+    void set_weights(const std::vector<value_type>& win, const std::vector<value_type>& wout) {
```

Comment:
A comment here and/or a change in the method name would help to indicate that these weights correspond to the contribution of the default values, which may not be the CV area when there are ion-concentration writing mechanisms present.

Reply:
done: added a comment.